### PR TITLE
Add class documentation for types

### PR DESCRIPTION
**Pull request by matthiasbeyer**
This PR adds class documentation for all types. It is ready to get merged.

### CODE REVIEW

File: src/compositor/framebuffer_device.h [hunk 1]
```diff
@@ -48,12 +48,14 @@
...
+ * @extends ws_object
+ *
  * The ws_framebuffer
  */
 struct ws_framebuffer_device {
-    struct ws_object obj; //!< Supertype
-    int fd; //!< The filedescriptor
-    char* path; //!< The path to the device file
+    struct ws_object obj; //!< @protected Base class.
+    int fd; //!< @protected The filedescriptor
```
Comment (TheNeikos) [inline src/compositor/framebuffer_device.h]: This is public, but read only.

File: src/compositor/framebuffer_device.h [hunk 2]
```diff
@@ -48,12 +48,14 @@
...
+    char* path; //!< @protected The path to the device file
```
Comment (TheNeikos) [inline src/compositor/framebuffer_device.h]: Same thing here

File: src/compositor/monitor.h [hunk 3]
```diff
@@ -50,23 +50,25 @@
...
+    struct ws_framebuffer_device* fb_dev; //!< @protected Framebuffer Device
+    drmModeModeInfo mode; //!< @protected mode of the monitor
+    uint32_t handle; //!< @protected Handle of the frame buffer
+    uint32_t fb; //!< @protected id of the frame buffer
+    uint32_t conn; //!< @protected id of the connector
+    uint32_t crtc; //!< @protected id of the "monitor"
+    drmModeCrtc* saved_crtc; //!< @protected drm internal datastructure for crtc
 
-    struct ws_set surfaces;
+    struct ws_set surfaces; //!< @protected
```
Comment (TheNeikos) [inline src/compositor/monitor.h]: None of these are protected, but public

**Comment by matthiasbeyer** [comment]
I'm ready for merging here. @TheNeikos , @xVin , maybe @neithernut ?

**Comment by TheNeikos** [comment]
From what I saw, it was good

**Comment by xVin** [comment]
Same for me.

**Pull request merged**